tools/trace.py

```diff
@@ -20,6 +20,9 @@
 import traceback
 import sys
 
+if sys.version_info >= (3, 0):
+    xrange=range
```

Comment:
Sure, something along those lines would be good, although the intended list is meant to hold indices, so it might need some massaging.